README.md

```diff
@@ -0,0 +1,83 @@
+# nightfall_dlp
+![nightfalldlp](https://www.finsmes.com/wp-content/uploads/2019/11/Nightfall-AI.png "nightfalldlp")
+### NightfallDLP - a code review tool that protects you from committing sensitive information
+
+NightfallDLP scans your commits for secrets or sensitive information and posts review comments to your code hosting 
+services automatically. It’s intended to be used as a part of your CI to simplify the development process, improve your 
+security, and ensure you never accidentally leak secrets or other sensitive information via an accidental commit.
+
+## Supported CI Services
+### GithubActions
+Example using the NightfallDLP Github Action inside a Github Workflow  
+_(**N.B.** you must use the actions/checkout step before the running the nightfalldlp action in order for it to function properly)_
+```yaml
+name: nightfalldlp
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+jobs:
+  run-nightfalldlp:
+    name: nightfalldlp
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repo Action
+        uses: actions/checkout@v2
+
+      - name: nightfallDLP action step
+        uses: nightfallai/nightfall_dlp_action@CreateAction
+        env:
+          NIGHTFALL_API_KEY: ${{ secrets.NIGHTFALL_API_KEY }}
+          EVENT_BEFORE: ${{ github.event.before }}
+```
+
+### Action Spec  
+NightfallDLP requires a config file and a few Environment variables in order to run  
+
+**Config File (detectors)**  
+ - place a `.nightfalldlp/` directory within the root of your target repository, and inside it a `config.json` file
+ in which you can configure your detectors (see `Detectors` section below for more information on Detectors)  
+ - inside the `detectors` map the keys are Detector Names and the values are the minimum likelihood for a detector to be triggered 
+
+ - sample `.nightfalldlp/config.json` file
+```json
+{
+    "detectors": {
+        "CREDIT_CARD_NUMBER": "POSSIBLE",
+        "PHONE_NUMBER": "LIKELY"
+    }
+}
+```
+**Env Variables**      
+These variables should be made available to the nightfall_dlp_action by adding them to the `env:` key in your workflow  
+1) `NIGHTFALL_API_KEY`
+    - get a (FREE) Nightfall AI DLP Scan API Key by registering an account with the Nightfall API HERE
+    - add this variable to your target repository's "Github Secrets" and passed in to your Github Workflow's `env`.
+2) `GITHUB_TOKEN`
+    - this is automatically injected by Github inside each Workflow (via the `secrets` context), you just need to set it 
+    to the env key
```

Comment:
Do you think we should provide a reason for requiring this? Since this token will give us access to their repo.

Ex. `This token is required to determine which files have been updated`.

Reply:
Hmm good question. I'll add it for now and we can see what folks think :)